PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -81,10 +82,27 @@ def __is_valid_seed(seed: bytes) -> bool:
 
 def _secure_seed_compare(expected_seed: bytes, actual_seed: bytes) -> bool:
     """Compare provided seeds in constant time to prevent timing attacks.
+    :raises TypeError: if parameters are not bytes-like objects
     :rtype: bool
     :return: True if seeds are the same, False otherwise.
     """
-    pass
+    # TODO: Can we use `hmac` module?
+    # > hmac.compare_digest` uses an approach designed to prevent timing
+    # > analysis by avoiding content-based short circuiting behaviour, making
+    #  > it appropriate for cryptography
+    # > https://docs.python.org/3.7/library/hmac.html#hmac.compare_digest
+    #
+    # > Note: If a and b are of different lengths, or if an error occurs,
+    # > a timing attack could theoretically reveal information about the types
+    #  > and lengths of a and b—but not their values.
+    #
+    # Type and length of seeds is known to the attacker, but not the value of expected seed.
+    if not (type(expected_seed) == type(actual_seed) == bytes):
```

Comment:
seems to me this should be covered by __is_valid_seed, either here or before the function is called. Or is this different?

Reply:
Function `__is_valid_seed` currently does not check types during runtime. Type hints are for static analysis, but are not enforced during runtime. This is here because of `hmac.compare_digest`, as written in the comments.

However, there would be benefits of using entropy, mnemonic, and seed as classes, but we can get to it in next steps. :)

Reply:
it does check types here https://github.com/mvondracek/PA193_mnemonic_Slytherin/pull/21

Reply:
OK, but that's not yet merged.

Reply:
Of course, we should unify our styles before changes are merged to dev though. Checks currently differ between pull requests in this sense.

Reply:
We can wait until #21 is merged, I will then sync changes.

Reply:
Would be solved by #26.